### PR TITLE
Stripy link

### LINKED ISSUE #3178: Customize STR variant page
Much info on the STR variantS page is unavailable on the variant page. The variant page for an STR variant also has a lot of empty cards with info that does not fit the variant type.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.
Fix #3178 and #3177: add back OMIM annotations, add STRIpy and GnomAD STR links for STR variants.
Also tentatively break out STR-variant page to its own to make it easier to layout when removing several unused non-STR cards. I dunno, we could just scale back that one; the former stuff is what the customers in question actually need, the rest is decorative.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/links.py [hunk 1]
```diff
@@ -63,6 +63,24 @@ def add_gene_links(gene_obj, build=37):
     gene_obj["cbioportal_link"] = cbioportal_gene(hgnc_symbol)
     gene_obj["civic_link"] = civic_gene(hgnc_symbol)
     gene_obj["iarctp53_link"] = iarctp53(hgnc_symbol)
+    gene_obj["stripy_link"] = stripy_gene(hgnc_symbol)
+    gene_obj["gnomad_str_link"] = gnomad_str_gene(hgnc_symbol)
+
+
+def stripy_gene(hgnc_symbol):
+    link = "https://stripy.org/database/{}"
```
Comment (northwestwitch) [inline scout/server/links.py:71]: This works 👍🏻

image

File: scout/server/links.py [hunk 2]
```diff
@@ -63,6 +63,24 @@ def add_gene_links(gene_obj, build=37):
...
+
+    if not hgnc_symbol:
+        return None
+    return link.format(hgnc_symbol)
+
+
+def gnomad_str_gene(hgnc_symbol):
+    link = "https://gnomad.broadinstitute.org/short-tandem-repeat/{}?dataset=gnomad_r3"
```
Comment (northwestwitch) [inline scout/server/links.py:79]: This one also: 👍🏻 

image

File: scout/server/blueprints/variants/templates/variants/str-variants.html [hunk 3]
```diff
@@ -216,4 +216,5 @@
   </div>"
   title="">{{ variant.str_repid }}
   </a>
+  {{ variant.stripy_link }}
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/templates/variants/str-variants.html:219]: Where does this change reflect on the STR variantS page?

File: scout/server/blueprints/variant/utils.py [hunk 4]
```diff
@@ -60,16 +60,21 @@ def update_representative_gene(variant_obj, variant_genes):
...
-            variant_genes, key=lambda gn: SO_TERMS[gn["functional_annotation"]]["rank"]
+            variant_genes,
+            key=lambda gn: SO_TERMS.get(
+                gn.get("functional_annotation"), {"rank": 999, "region": "unknown"}
+            )["rank"],
         )
         # get HGVNp identifier from the canonical transcript
         hgvsp_identifier = None
-        for tc in first_rep_gene["transcripts"]:
+        for tc in first_rep_gene.get("transcripts", []):
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/utils.py:73]: 👍🏻

File: scout/server/blueprints/variant/utils.py [hunk 5]
```diff
@@ -180,6 +185,14 @@ def add_gene_info(store, variant_obj, gene_panels=None, genome_build=None):
     variant_obj["has_refseq"] = False
     variant_obj["disease_associated_transcripts"] = []
     all_models = set()
+
+    # seed genes structure for (STR) variants that have only hgnc_ids
+    if not variant_obj.get("genes") and variant_obj.get("hgnc_ids"):
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/utils.py:190]: I guess this is why the gene name reappears on the STR variant page right?

Main branch:

image


This branch:

image

**Review by northwestwitch — APPROVED** [review]
This works!

- Gene names are shown on the STR variant page
- STRipy and GNOMAD strs buttons point to the relative resources

I just couldn't figure out what is the change in str-variants.html page, like I can't see any difference in the page.. 🤔 

Check the Vulture comment for the pp import that can be removed!

File: scout/server/blueprints/variants/templates/variants/str-variants.html (hunk 3 above)
Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/str-variants.html:219]: Ahem, thank you - not sure it does! It was supposed to be in the variantS popover for the locus name, but I left it unfinished when deciding to add it to the variant page directly. Will fix!

File: scout/server/blueprints/variant/utils.py (hunk 5 above)
Comment (dnil) [reply scout/server/blueprints/variant/utils.py:190]: Yes, and especially why the OMIM info gets populated. There could still be loci that dont have an associated HGNC-id but for now I believe this covers all the clinical ones.

File: scout/server/blueprints/variants/templates/variants/str-variants.html (hunk 3 above)
Comment (dnil) [reply scout/server/blueprints/variants/templates/variants/str-variants.html]: Oki, now it looks like this:
Screenshot 2022-02-28 at 15 40 03
with the S-badge link going to STRipy and the G-one to GnomAD. And OMIM disease info is also in the popover.

**Pull request merged**